### ISSUE TITLE
Waiting for multiple promises to complete?

### ISSUE THREAD

**Issue by bibhas2** [mxcl/PromiseKit#352]
I do not see any way for multiple promises to complete. We need that to run a couple of tasks to run in parallel and then wait for all of them to finish.

**Comment by wangii** [comment]
Isn't it "when" designed for?

**Comment by nathanhosselton** [comment]
See: [`when`](http://promisekit.org/when/) (or alternatively: [`join`](https://github.com/mxcl/PromiseKit/blob/master/Sources/join.swift))

**Comment by bibhas2** [comment]
OK, cool ``when`` will do the trick. It will be great to see a mention or link from the chaining section of the doc. Right now chaining only mentions sequential calls. Thank you all!

**Comment by FruitAndAShape** [comment]
What does one do if the promises are of a different type?

**Comment by mxcl** [comment]
Open new ticket.
